lib/builder_web.ml

```diff
@@ -627,6 +694,8 @@ let routes ~datadir ~cachedir ~configdir ~expired_jobs =
     `Get, "/job/:job/failed", (w job_with_failed);
     `Get, "/job/:job/build/latest/**", (w redirect_latest);
     `Get, "/job/:job/build/latest", (w redirect_latest_no_slash);
+    `Get, "/job/:job/build/latest-uuid", (w return_latest_uuid);
+    `Get, "/search-uuid", (w return_uuid);
```

Comment:
I think I'm maybe more inclined to have "/job/:job/build/:build" return a generic build info object with uuid, (job,) platform when there's an `Accept: application/json` or so. And then you would use the existing "/hash" and "/job/:job/build/latest" endpoints and follow the redirects.